pkg/products/monitoring/reconciler.go

```diff
@@ -569,6 +574,16 @@ func (r *Reconciler) reconcileComponents(ctx context.Context, serverClient k8scl
 			PrometheusInstanceNamespaces:     r.Config.GetOperatorNamespace(),
 			SelfSignedCerts:                  r.installation.Spec.SelfSignedCerts,
 		}
+
+		if isMultiAZCluster {
+			m.Spec.Affinity = resources.SelectAntiAffinityForCluster(isMultiAZCluster, map[string]string{
```

Comment:
```suggestion
			m.Spec.Affinity = resources.SelectAntiAffinityForCluster(antiAffinityRequired, map[string]string{
```

We still need to use the `antiAffinityRequired` value (from the env var) for this parameter. Otherwise the affinity rules will be set to required for every Multi AZ cluster

Reply:
I've made the update as per your suggestion. Thanks Sergio.